packages/fxa-auth-server/fxa-oauth-server/lib/config.js

```diff
@@ -396,34 +418,52 @@ var options = {
 
 conf.validate(options);
 
-// Replace openid key if file specified
+// Load our various keys from files if specified,
+// resolving paths to absolute form to remove any ambiguity.
+
 if (conf.get('openid.keyFile')) {
-  conf.set('openid.key', require(conf.get('openid.keyFile')));
+  const keyFile = path.resolve(__dirname, conf.get('openid.keyFile'));
+  conf.set('openid.keyFile', keyFile);
+  // If the file doesnt exist, or contains an empty object, then there's no active key.
+  conf.set('openid.key', null);
+  if (fs.existsSync(keyFile)) {
+    const key = JSON.parse(fs.readFileSync(keyFile));
+    if (key && Object.keys(key).length > 0) {
+      conf.set('openid.key', key);
+    }
+  }
+} else if (Object.keys(conf.get('openid.key')).length === 0) {
+  conf.set('openid.key', null);
 }
 
-if (conf.get('openid.oldKeyFile')) {
-  conf.set('openid.oldKey', require(conf.get('openid.oldKeyFile')));
+if (conf.get('openid.newKeyFile')) {
+  const newKeyFile = path.resolve(__dirname, conf.get('openid.newKeyFile'));
+  conf.set('openid.newKeyFile', newKeyFile);
+  // If the file doesnt exist, or contains an empty object, then there's no new key.
+  conf.set('openid.newKey', null);
+  if (fs.existsSync(newKeyFile)) {
+    const newKey = JSON.parse(fs.readFileSync(newKeyFile));
+    if (newKey && Object.keys(newKey).length > 0) {
+      conf.set('openid.newKey', newKey);
+    }
+  }
+} else if (Object.keys(conf.get('openid.newKey')).length === 0) {
+  conf.set('openid.newKey', null);
 }
 
-var key = conf.get('openid.key');
-assert.equal(key.kty, 'RSA', 'openid.key.kty must be RSA');
-assert(key.kid, 'openid.key.kid is required');
-assert(key.n, 'openid.key.n is required');
-assert(key.e, 'openid.key.e is required');
-assert(key.d, 'openid.key.d is required');
-
-var oldKey = conf.get('openid.oldKey');
-if (Object.keys(oldKey).length) {
-  assert.equal(oldKey.kty, 'RSA', 'openid.oldKey.kty must be RSA');
-  assert(oldKey.kid, 'openid.oldKey.kid is required');
-  assert.notEqual(
-    key.kid,
-    oldKey.kid,
-    'openid.key.kid must differ from oldKey'
-  );
-  assert(oldKey.n, 'openid.oldKey.n is required');
-  assert(oldKey.e, 'openid.oldKey.e is required');
-  assert(!oldKey.d, 'openid.oldKey.d is forbidden');
```

Comment:
I moved a bunch of this validation logic into `keys.js`, along with a Joi schema for doing it.